### PR TITLE
Clarifying escaping double quotes (re: #259)

### PR DESCRIPTION
**Pull request by talsraviv**

### CODE REVIEW

**Comment by talsraviv** [comment]
cc @moollaza @zekiel
Figured better to be explicit and keep it simple for now

File: duckduckhack/goodie/goodie_cheat_sheets.md [hunk 1]
```diff
@@ -91,11 +91,15 @@ The above Instant Answer was created by simply adding `vim.json`, summarized bel
...
 Brackets, `[ ]`, or braces, `{ }`, are used to wrap key combinations in code blocks. For convenience, if you include no brackets or braces, the entire string will be shown in a code block.
 
-To express literal brackets, braces, and backslash, escape with a double backslash: e.g. `{Ctrl + \\[}`.
+#### Escaping Special Characters
 
-In the case of trying to express a literal backslash you need to have a double backslash and each of them need to be escaped: e.g. `[Ctrl + \\\\]`
+Your cheat sheet might include characters which are themselves part of JSON syntax. To express these literally, you can *escape* them as follows:
 
-An uneven number of sequential backslashes will throw a JSON parsing error
+- To express a double quote, use a double backslash: `\\"`
```
Comment (moollaza) [inline duckduckhack/goodie/goodie_cheat_sheets.md]: This isn't quite right -- there's two levels of escaping that can go on in one of the JSON files.

As per the JSON Spec, there are certain chars which MUST be escaped to produce valid JSON. If you look at the 3rd diagram here, http://json.org/, you'll see all the valid chars that can come after a `\` (when the `\` is part of a JSON string)

e.g. `"`, `n`, `/`

This must always be followed in any JSON document, anywhere.

-----

Separately, we have our own "CheatSheet" syntax which looks for `[]` and `{}` and replaces them with `<code></code>`.

Now, not every bracket inside `key` string is meant to indicate the beginning/end of a code snippet and so we tell you to signify "literal" brackets in the string by "escaping" them with a `\`.

Seeing as `\` is the JSON escape character, you actually need to escape the `\` so it get's passed on to our secondary "CheatSheet" parser.

Thus `"{Ctrl + \\[}"` goes through the JSON parse and becomes `"{Ctrl + \[}"` which goes through the CheatSheet parser to become `<code>Ctrl + [</code>`.

Does that clarify things? It's kinda difficult to grasp what exactly is going on here.

I suppose we could "dumb it down" and say that you need to escape brackets in your string with `\\` and not bother explaining how this gets interpreted?

Comment (moollaza) [inline duckduckhack/goodie/goodie_cheat_sheets.md]: Basically if you want a literal quote, slash, tab, newline, etc it needs to be escaped with a single `\`

If you want a literal bracket, it needs to be escaped with a `\\`.

Comment (talsraviv) [inline duckduckhack/goodie/goodie_cheat_sheets.md]: @moollaza yeah - that makes a lot of sense explained that way. Check out what I did

Given the target audience of the cheat sheets doc, I've tried to spell things out as much as I can and keep the doc simple yet logical for any advanced readers.

**Comment by talsraviv** [comment]
@moollaza yeah that makes sense explained that way

Given the "low floor" of the the cheat sheet doc, I tried to keep it simple for beginners but logical for advanced users.

File: duckduckhack/goodie/goodie_cheat_sheets.md [hunk 2]
```diff
@@ -91,11 +91,21 @@ The above Instant Answer was created by simply adding `vim.json`, summarized bel
...
 
-To express literal brackets, braces, and backslash, escape with a double backslash: e.g. `{Ctrl + \\[}`.
+#### Escaping Special Characters
 
-In the case of trying to express a literal backslash you need to have a double backslash and each of them need to be escaped: e.g. `[Ctrl + \\\\]`
+Your cheat sheet might include characters which are themselves part of JSON syntax. To express these literally, escape them using backslashes, like [standard JSON](http://json.org):
 
-An uneven number of sequential backslashes will throw a JSON parsing error
+- To express a double quote, use a single backslash: `\"`
+- To express a forward slash, use a single backslash: `"\/"`
```
Comment (moollaza) [inline duckduckhack/goodie/goodie_cheat_sheets.md]: I think the double quotes here are a little confusing, `\/` should be fine.

Comment (moollaza) [inline duckduckhack/goodie/goodie_cheat_sheets.md]: Or maybe we should give examples with a little more text, for the sake of context? Happy either way.

File: duckduckhack/goodie/goodie_cheat_sheets.md [hunk 3]
```diff
@@ -91,11 +91,21 @@ The above Instant Answer was created by simply adding `vim.json`, summarized bel
...
+- For full list of characters, see the diagram on the right on [the official JSON documentation](http://json.org).
+
+Because cheat sheets display brackets `[ ]` and braces `{ }` as code blocks, express those literally using a **double backslash**:
+
+- If you want to express a literal bracket, use a double backslash `{Ctrl + \\[}`.
+- If you want to express a literal brace, use a double backslash `[Ctrl + \\{]`.
+- To express a **single literal backslash**, type four backslashes in a row: `[Ctrl + \\\\]`.
+
+*Note that an uneven number of sequential backslashes will throw an error.*
```
Comment (moollaza) [inline duckduckhack/goodie/goodie_cheat_sheets.md]: Not sure if this now seems contradictory to what we mentioned above? Perhaps we should indicate that this applies to 3+ (unless that's what sequential implies?)

If you think it get's the point across, I'm happy.

Comment (talsraviv) [inline duckduckhack/goodie/goodie_cheat_sheets.md]: Good point. 
Re-read it, and I think it's both accurate and clear.
